src/parser/shared/modules/features/STAT.tsx

```diff
@@ -13,26 +13,41 @@ import StaminaIcon from 'interface/icons/Stamina';
 import StrengthIcon from 'interface/icons/Strength';
 import VersatilityIcon from 'interface/icons/Versatility';
 
-enum STAT {
-  HEALTH = 'health',
-  STAMINA = 'stamina',
-  MANA = 'mana',
+export enum PRIMARY_STAT {
   STRENGTH = 'strength',
   AGILITY = 'agility',
   INTELLECT = 'intellect',
+}
+
+export enum SECONDARY_STAT {
   CRITICAL_STRIKE = 'criticalstrike',
   HASTE = 'haste',
-  HASTE_HPCT = 'hastehpct',
-  HASTE_HPM = 'hastehpm',
   MASTERY = 'mastery',
   VERSATILITY = 'versatility',
+}
+
+export enum OTHER_STAT {
+  HEALTH = 'health',
+  STAMINA = 'stamina',
+  MANA = 'mana',
+  HASTE_HPCT = 'hastehpct',
+  HASTE_HPM = 'hastehpm',
   VERSATILITY_DR = 'versatilitydr',
   LEECH = 'leech',
   AVOIDANCE = 'avoidance',
   SPEED = 'speed',
   UNKNOWN = 'unknown',
 }
 
+type STAT = PRIMARY_STAT | SECONDARY_STAT | OTHER_STAT;
```

Comment:
I'd recommend declaring this type as `Stat` and then removing the `eslint-disable-next-line` below it.

Reply:
I figured it was to remain as a enum functionally, but maybe it's just better to be clear between type and const

Reply:
Supposing that you do the below, you'll get an error about `STAT` being a type but is being used as a value:
```
const something = STAT.HEALTH;
```

Unfortunately, can't union enums and have the union work as a directly addressable enum.

Reply:
You can't union enums, but on the rows below this I declare a const as well. It's possible to have a variable which will be used when used as a value, and a type which will be used when used as a type. So pretty much like an enum. 

```
// type used as type and object used as value
const myStat: STAT = STAT.STRENGTH;
```

But I can understand that it's clearer to have two different names for them.

Reply:
There's a reason the no-redeclare rule exists, imo it's bad design if one cannot discern which is which easily, so I'd vote for different name. Personally, I'm not a big fan of going for a `Stat` though as then it's just same word but different casing changes the meaning.
I'd probably personally defer to something like `STAT_TYPES`, but it might even be that the const should be the with the different name rather than the type, might need to see how Stat is used in this module as is

Reply:
i totally missed this when reviewing. I agree that we don't want to redeclare the type as a const.

it may not be the prettiest, but since `enum`s are objects at runtime, perhaps just...duplicate the values? these things don't change very often.

that is: go back to `enum STAT` and add an `enum PRIMARY_STAT` that includes the 3 primary stats. the `SECONDARY_STAT` and `OTHER_STAT` values are new and not currently used.

Reply:
> i totally missed this when reviewing. I agree that we don't want to redeclare the type as a const.
> 
> it may not be the prettiest, but since `enum`s are objects at runtime, perhaps just...duplicate the values? these things don't change very often.
> 
> that is: go back to `enum STAT` and add an `enum PRIMARY_STAT` that includes the 3 primary stats. the `SECONDARY_STAT` and `OTHER_STAT` values are new and not currently used.

That's fair 😅

Reply:
There we go, kept the original `enum`, then create two new enums for subset of main `STAT` enum. 

There's no overlapping type/value, still works.

Reply:
Actually I take it back.. the solution with multiple enums isn't great either because the types `STAT !== PRIMARY_STAT`... So if you have a function that expect `STAT` you cannot pass `PRIMARY_STAT.STRENGTH` eventhough the string is `strength`.

The `type STAT; const STAT` let's us build an object with enum values and a type which is a union of those enums. That way `PRIMARY_STAT.STRENGTH` satisfies `PRIMARY_STAT` _and_ `STAT`. `PRIMARY_STAT.STRENGTH === STAT.STRENGTH` will also be true.

Having a value and type with the same name is [perfectly legal](https://www.typescriptlang.org/docs/handbook/declaration-files/deep-dive.html#simple-combinations-one-name-multiple-meanings), an `enum` is actually both a value and type to begin with. Also looking at the eslint-rule in particular, it even has a [recommendation](https://typescript-eslint.io/rules/no-redeclare/) to not be enabled since Typescript itself will catch redeclarations.

If we still don't like this approach, I can only see two ways. 

1. Separate the value and type by name, as suggested originally. Either separate sub-values `enum STAT; const PrimaryStat; type PRIMARY_STAT;` _or_ separate main Stat `const STAT; type Stat; enum PRIMARY_STAT`. 
2. Have a `enum STAT` and then `const PRIMARY_STAT = { STRENGTH: STAT.STRENGTH }` but _no_ type for the primary/secondary subsets.

Reply:
given the documentation and the other quirks of TS in this area, I'm okay with using the double-namespace to have both exist with the same name.